### DISCUSSION TITLE
Is it possible to tell if cypress is running via open or run in the tests/commands file.

### DISCUSSION THREAD

**Discussion by sl33t** [cypress-io/cypress#14296] [Questions and Help] [answered]
I have a command I only want to trigger when cypress is in run mode and I want to ignore it in open mode. Currently I am using a custom script and an environment variable for this. I would like to do this in a native way if possible but cannot find this in the docs.

**Comment by bahmutov** [comment]
You can find the answer in https://glebbahmutov.com/blog/cypress-tips-and-tricks/#interactive-and-headed-mode but we probably should include this information in the official docs

**Reply by sl33t** [reply]
Thanks! Exactly what I wanted! Please do add it to the docs.

**Reply by sl33t** [reply]
Typescript does not like Cypress.config('isInteractive') maybe this needs to be added to type definitions as well.

**Reply by bahmutov** [reply]
I have added this to the docs here https://docs.cypress.io/guides/references/configuration.html#isInteractive still need to add some of the run-time config properties to the typescript definitions

**Reply by bahmutov** [reply]
To fill the typescript definitions I have opened https://github.com/cypress-io/cypress/issues/14392

**Comment by estefafdez** [comment]
what would you say is the best way to control the open/run commands along the browsers and environments?

**Reply by bahmutov** [reply]
Not sure I understand this question. Do you mean to vary the browser depending on the environment? Or run specific test commands depending on the environment?

**Comment by jennifer-shehane** [comment, marked as answer]
Use `Cypress.config('isInteractive')`, true under `cypress open`, false under `cypress run`. Available in specs and support files.
